### PR TITLE
Cleanup: Config Cleanup

### PR DESCRIPTION
**Pull request by dislogical**
This moves most logic in Config out into the module scope, which means that you don't have to qualify all of your function calls with `Config`, and you can use UFCS to call like this: `node.tryFind( "My.Path", myPath )`.

Requires [`feature/ConfigCleanup`](https://github.com/Circular-Studios/Sample-Dash-Game/compare/develop...feature%2FConfigCleanup) on Sample Dash Game.

### CODE REVIEW

**Comment by wozniakty** [comment]
:+1:

File: source/utility/config.d [hunk 1]
```diff
@@ -130,8 +202,259 @@ Node loadYamlFile( string filePath )
...
+
+    return true;
+}
+
+/// ditto
+final bool tryFind( T = Node )( Node node, string path, ref Variant result ) nothrow @safe
+{
+    // Get the value
+    T temp;
+    bool found = tryFind( path, temp, node );
```
Comment (brooklynlittell) [inline source/utility/config.d]: Is the ordering for this call correct? All three tryFind definitions are in the order `node, path, result`

**Comment by brooklynlittell** [comment]
With ae5bbf2 it looks good to go :+1:

**Comment by ericchristenson** [comment]
:100: looks good

**Pull request merged**